packages/style-engine/class-wp-style-engine-css-rule.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Rule
+ *
+ * An object for CSS rules.
+ *
+ * @package Gutenberg
+ */
+
+if ( class_exists( 'WP_Style_Engine_CSS_Rule' ) ) {
+	return;
+}
+
+/**
+ * Holds, sanitizes, processes and prints CSS declarations for the style engine.
+ *
+ * @access private
+ */
+class WP_Style_Engine_CSS_Rule {
+
+	/**
+	 * The selector.
+	 *
+	 * @var string
+	 */
+	protected $selector;
+
+	/**
+	 * The selector declarations.
+	 *
+	 * Contains an array of WP_Style_Engine_CSS_Declarations objects.
+	 *
+	 * @var WP_Style_Engine_CSS_Declarations[]
+	 */
+	protected $declarations = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @param string                             $selector     The CSS selector.
+	 * @param WP_Style_Engine_CSS_Declarations[] $declarations An array of WP_Style_Engine_CSS_Declarations objects.
+	 */
+	public function __construct( $selector = '', $declarations = array() ) {
+		$this->set_selector( $selector );
+		$this->set_declarations( $declarations );
+	}
+
+	/**
+	 * Set the selector.
+	 *
+	 * @param string $selector The CSS selector.
+	 */
+	public function set_selector( $selector ) {
+		if ( empty( $selector ) ) {
+			return;
+		}
+		$this->selector = esc_html( $selector );
```

Comment:
This is a slightly tricky issue, but how might we deal with the child combinator in a selector (`>`) without escaping it to an html character? It's used in a couple of places, but primarily the `flow` layout in global styles and in the `layout.php` block support (e.g. `$selector > *` and `$selector > * + *`.

Reply:
Maybe we create our own version of https://developer.wordpress.org/reference/functions/sanitize_key/ ?

Reply:
You're right, using `esc_html` here would be an issue.
Considering these selectors are not provided by users and are not dependent on any kind of user-input, I removed the `esc_html` call here since it won't cause any security issues. It was mostly a force of habit (escape everything) 😅 
The Declarations class uses the safe-css methods to remove any unwanted properties, but the final CSS should be escaped on output - which means that it should be in the Output object when that is created 👍